sdks/python/apache_beam/transforms/util.py

```diff
@@ -864,3 +867,225 @@ def add_window_info(element, timestamp=DoFn.TimestampParam,
+          results.append(match.group(groupNum + 1))
+      if results:
+        yield results
+    return pcoll | FlatMap(_process)
+
+  @staticmethod
+  @ptransform_fn
+  def _matches_kv_object(pcoll, regex, keyGroup, valueGroup):
+    """
+    PTransfrom to find the matches as per regex and return as a KV pair.
+    """
+    def _process(element):
+      match = regex.match(element)
+      if match:
+        yield (match.group(keyGroup), match.group(valueGroup))
+
+    return pcoll | FlatMap(_process)
+
+  @staticmethod
+  @typehints.with_input_types(str)
+  @typehints.with_output_types(str)
+  @ptransform_fn
+  def matches(pcoll, regex, group=None):
+    """
+    Returns the matches if the entire line matched the Regex. Returns the
+    entire line (group 0 by default). Group can be integer value and a string
+    value.
+
+    Args:
+      regex: the regular expression string or (re.compile) pattern.
+      group: (optional) name of the group, it can be integer or a string value.
+    """
+    regex = Regex._regex_compile(regex)
+    group = group or 0
+
+    def _process(element):
+      m = regex.match(element)
+      if m:
+        yield m.group(group)
+    return pcoll | FlatMap(_process)
+
+  @staticmethod
+  @typehints.with_input_types(str)
+  @typehints.with_output_types(str)
+  @ptransform_fn
+  def all_matches(pcoll, regex):
+    """
+    Returns the matches if the entire line matches the Regex. Returns all
+    groups.
+
+    Args:
+      regex: the regular expression string or (re.compile) pattern.
+    """
+    regex = Regex._regex_compile(regex)
+    return pcoll | Regex._matches_all_object(regex=regex)
+
+  @staticmethod
+  @typehints.with_input_types(str)
+  @typehints.with_output_types(typehints.KV[K, V])
+  @ptransform_fn
+  def matches_kv(pcoll, regex, keyGroup, valueGroup):
+    """
+    Returns the matches if the entire line matches the Regex. Returns the
+    specified groups as the key and value pair.
+
+    Args:
+      regex: the regular expression string or (re.compile) pattern.
+      keyGroup: The Regex group to use as the key. Can be int or str.
+      valueGroup: The Regex group to use the value. Can be int or str.
+    """
+    regex = Regex._regex_compile(regex)
+    return pcoll | Regex._matches_kv_object(regex=regex, keyGroup=keyGroup,
+                                            valueGroup=valueGroup)
+
+  @staticmethod
+  @typehints.with_input_types(str)
+  @typehints.with_output_types(str)
+  @ptransform_fn
+  def find(pcoll, regex, group=None):
+    """
+    Returns the matches if a portion of the line matches the Regex. Returns
+    the entire line (group 0 by default). Group can be integer value and a
+    string value.
+
+    Args:
+      regex: the regular expression string or (re.compile) pattern.
+      group: (optional) name of the group, it can be integer or a string value.
+    """
+    regex = Regex._regex_compile(regex)
+    group = group or 0
```

Comment:
Why not just set 0 as the default value above?

Reply:
Done